File: score-board-client/src/app/models/player.ts
import {Role} from './role';
import {Settings} from './settings';

export class Player {
  id: string;
  playerName: string;
  firstName?: string;
  lastName?: string;
  email?: string;
  password: string;
  roleName: string;
  role: Role;
  mustChangePassword?: boolean;
  isActive?: boolean;
  lastLogin?: Date;
  registrationDate?: Date;
  activationDate?: Date;
  image: string;
  token?: string;
  settings?: Settings;
}

File: score-board-server/ScoreBoard.Domain/Models/Player.cs
using System;
using PlayerRole = ScoreBoard.Domain.Enums.PlayerRole;

namespace ScoreBoard.Domain.Models
{
    public class Player
    {
        public Player()
        {
            Id = Guid.NewGuid();
            Role = PlayerRole.Guest;
        }

        public Guid Id { get; set; }
        public string PlayerName { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }
        public byte[] PasswordHash { get; set; }
        public byte[] PasswordSalt { get; set; }
        public bool IsActive { get; set; }
        public bool MustChangePassword { get; set; }
        public string Image { get; set; }
        public PlayerRole Role { get; set; }
        public DateTime? LastLogin { get; set; }
        public DateTime? RegistrationDate { get; set; }
        public DateTime? ActivationDate { get; set; }
        public Settings Settings { get; set; }
    }
}
